site/public/js/details.js

```diff
@@ -129,3 +131,37 @@ function inquiry_update() {
         button.textContent = 'Grade Inquiry Only: Off';
     }
 }
+
+function checkProperTicks() {
+    fillAllCheckboxes(true);
+    const selectedColumns = JSON.parse(decodeURIComponent(Cookies.get('grading_details_columns')));
+    for (let i = 0; i < selectedColumns.length; i++) {
+        document.getElementById(`toggle-${selectedColumns[i]}`).checked = false;
+    }
+}
+
+function toggleColumnsForm() {
+    showPopup('#toggle-columns-form');
+    checkProperTicks();
+}
+
+function fillAllCheckboxes(val) {
+    for (let i = 0; i < checkboxes.length; i++) {
+        checkboxes[i].checked = val;
+    }
+}
+
+function updateGradingDetailsColumns() {
+    getCheckboxValues();
+    location.reload();
```

Comment:
Do you actually need to reload here, or could you just toggle classes for each column to hide them without reloading?  I can see good arguments both for and against doing this.  Thoughts? 🤔